src/kernel/include/inv.h

```diff
@@ -90,6 +90,26 @@ sret_deactivate(struct cap_captbl *t, capid_t capin, livenessid_t lid)
 { return cap_capdeactivate(t, capin, CAP_SRET, lid); }
 
 static int
+asnd_construct(struct cap_asnd *asndc, struct cap_arcv *arcvc, capid_t rcv_cap, u32_t budget, u32_t period)
```

Comment:
there might have been a miscommunication.  The goal is to avoid replication where possible.  I don't see a good reason why we can't do that here.  *After* we resolve the `struct cap_asnd` in `asnd_activate`, it should be able to call this (or something like this), right?  It is literally a copy and paste so I don't understand your justification.

You need to think through ways to make sure that you're creating maintainable, reasonable, code.